### ISSUE TITLE
template_cloudinit_config needs two applys to propagate a change

### ISSUE THREAD

**Issue by wr0ngway** [hashicorp/terraform#6899]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.
### Terraform Version

0.6.16
### Affected Resource(s)
- template_cloudinit_config
### Terraform Configuration Files

``` hcl
resource "template_file" "test" {
  template = "# foo ${bar} baz"
  vars {
    bar = "howdy"
  }
}

resource "template_cloudinit_config" "test" {
  gzip = false
  base64_encode = false

  part {
    content_type = "text/cloud-config"
    content = "${template_file.test.rendered}"
    merge_type = "list(append)+dict(recurse_array)+str()"
  }
}

resource "aws_instance" "test" {
  ami = "ami-5c207736"
  instance_type = "m3.medium"

  user_data = "${template_cloudinit_config.test.rendered}"
}
```
### Expected Behavior

The second `terraform apply` should change all three resources.
### Actual Behavior

Running `terraform apply` the first time does the right thing.  Then change the value of bar variable in template file.  When doing the second `terraform apply`,  only changes template_file and template_cloudinit_config, but change doesn't propagate to the aws_instance that is using the template_cloudinit_config.  A third `terraform apply` picks up that change.
### Steps to Reproduce
1. `terraform apply` All three resources get created
2. Change the value of bar variable in template_file 
3. `terraform apply` only changes template_file and template_cloudinit_config, but change doesn't propagate to the aws_instance that is using the template_cloudinit_config
4. `terraform apply` The aws_instance finally gets modified

**Comment by jrnt30** [comment]
We too are seeing this behavior as well.  We are using a bit different config, but same results. 

Major differences are gzip/base64 & the lack of merge_type, neither of which should matter. 

```
resource "template_cloudinit_config" "cloud_init" {
  gzip          = true
  base64_encode = true

  part {
    filename = "init.cfg"
    content  = "${template_file.cloud_config.rendered}"
  }
}
```

**Comment by ColinHebert** [comment]
@jen20 I created a quick fix for this one. Would you mind trying to get it shipped with the 0.7 release?

**Comment by mitchellh** [comment]
Fixed with data sources!
